x-pack/plugins/security_solution/public/resolver/store/data/reducer.test.ts

```diff
@@ -22,14 +23,46 @@ describe('Resolver Data Middleware', () => {
   });
 
   describe('when data was received and the ancestry and children edges had cursors', () => {
+    let firstChildNodeInTree: TreeNode;
+    let statsForFirstChild: Record<string, number>;
     beforeEach(() => {
       const generator = new EndpointDocGenerator('seed');
+      const baseTree = generator.generateTree({
+        ancestors: 1,
+        generations: 2,
+        children: 3,
+        percentWithRelated: 100,
+        alwaysGenMaxChildrenPerNode: true,
+      });
+      const { children } = baseTree;
+      firstChildNodeInTree = [...children.values()][0];
+      statsForFirstChild = firstChildNodeInTree.relatedEvents.reduce(
+        (counts: Record<string, number>, relatedEvent) => {
+          for (const category of [relatedEvent.event.category].flat()) {
+            counts[category] = counts[category] ? counts[category] + 1 : 1;
+          }
+          return counts;
+        },
+        {}
+      );
       const tree = mockResolverTree({
-        events: generator.generateTree({ ancestors: 1, generations: 2, children: 2 }).allEvents,
+        events: baseTree.allEvents,
         cursors: {
-          childrenNextChild: 'aValidChildursor',
+          childrenNextChild: 'aValidChildCursor',
           ancestryNextAncestor: 'aValidAncestorCursor',
         },
+        children: [...baseTree.children.values()].map((node) => {
```

Comment:
Could you add a comment here explaining what this code is doing and how its used in the test?

Reply:
https://github.com/elastic/kibana/pull/70097/commits/836f8f9f340b15aed62d1b9bed83316ed6ee2ba5